### ISSUE TITLE
resolving() events on interfaces are called twice 

### ISSUE THREAD

**Issue by tomlankhorst** [laravel/framework#23699]
In L5.5, 5.6 I experience an issue with _resolving_ callbacks on dependencies registered in the container with an interface class as abstract. A demonstration:

Having some interface 

```php
IBar {}
```

and an implementation 

```php
Foo implements IBar {}
```

registered (here as a singleton) in the container 

```php
app()->singleton( IBar::class, Foo::class );
```

with a _resolving_ callback 

```php
app()->resolving( IBar::class, $callback );
```

fires `$callback` **twice** when the abstract is resolved!

This is probably because:

1. When the abstract `IBar` resolves, it triggers the callback
2. To resolve `IBar`, `Foo` must be constructed. As `Foo` implements `IBar`, it triggers the `resolving` event too! 

This problem can be omitted when avoiding Laravel's container to build Foo by registering a callback instead:

```php
app()->singleton( IBar::class, f(){ return new Foo; } )
```

But this abolishes the ease of automatic dependency injection. 

So... can we solve this issue by not firing the `resolving()` callback when we are building the implementation of an interface that the callback is registered for?

**Comment by sisve** [comment]
Slightly relevant: https://github.com/laravel/framework/pull/23290

**Comment by tomlankhorst** [comment]
Fine, I'll submit a PR these days that will address this issue.

**Comment by tomlankhorst** [comment]
"... these days..."
#23701

**Comment by taylorotwell** [comment]
Fixed in 5.8.

**Comment by hkodavid** [comment]
The bug is still in 5.8.12
